IPG-Automotive-UK/react-ui
# VE-204 Virtualised Autocomplete

Closes/Contributes [VE-204](https://sce.myjetbrains.com/youtrack/issue/VE-204/Virtualised-Autocomplete)

## Changes




**Changes in CheckboxFilter.tsx**:

- Enhanced functionality for the checkbox filter component.

- Used `react-window` for handling larger datasets.

**New File: VirtualizedAutocomplete.tsx**:

- Introduced a virtualized autocomplete component to handle large lists efficiently.

- Integrated virtualization to improve performance for rendering and searching within large dropdowns.

**Changes in AlwaysOpenAutocomplete**.tsx

- Refactored the component to ensure better performance and maintainability, especially with large datasets.

- Used `react-window` for handling larger datasets.

**Changes in CheckboxFilter.test**.tsx

- Removed the last 4 options ("Gate 1", "Gate 4", "Gate 5", "Gate 10") from the expected array because `VirtualizedAutocomplete` renders only the first 10 options initially for performance optimization.


**Preview** - Checkbox Filter `Default`
![Default](https://github.com/user-attachments/assets/8262d337-d2a7-403b-8f00-e1f7a8b87e0c)


**Preview** - Checkbox Filter `Always Open`
![AlwaysOpen](https://github.com/user-attachments/assets/8f40efb7-319f-4cd4-b35a-49a4875545e0)



## Dependencies



NA

## UI/UX



NA

## Testing notes



1. Open storybook
2. Check the `Filters/CheckboxFilter/Always Open` component
3. Make sure the component renders only the first 10 options initially, as per the virtualization logic
4. Scroll down within the dropdown and verify that additional options are rendered dynamically


## Author checklist

Before I request a review:
